connector/connect/server/src/main/scala/org/apache/spark/sql/connect/execution/ExecuteGrpcResponseSender.scala

```diff
@@ -201,9 +237,18 @@ private[connect] class ExecuteGrpcResponseSender[T <: Message](
           // The state of interrupted, response and lastIndex are changed under executionObserver
           // monitor, and will notify upon state change.
           if (response.isEmpty) {
-            val timeout = Math.max(1, deadlineTimeMillis - System.currentTimeMillis())
+            // Wake up more frequently to send the progress updates.
+            val progressTimeout =
+              executeHolder.sessionHolder.session.conf.get(CONNECT_PROGRESS_REPORT_INTERVAL)
+            // If the progress feature is disabled, wait for the deadline.
+            val timeout = if (progressTimeout > 0) {
+              progressTimeout
+            } else {
+              Math.max(1, deadlineTimeMillis - System.currentTimeMillis())
+            }
```

Comment:
nit:
```
            var timeout = Math.max(1, deadlineTimeMillis - System.currentTimeMillis())
            // Wake up more frequently to send the progress updates.
            val progressTimeout =
              executeHolder.sessionHolder.session.conf.get(CONNECT_PROGRESS_REPORT_INTERVAL)
            if (progressTimeout > 0) {
              Math.min(progressTimeout, timeout)
            }
```
otherwise, progressTimeout may make us wait beyond the deadline.

Reply:
https://github.com/apache/spark/pull/49474